src/live_restore/live_restore_server.c

```diff
@@ -181,48 +216,67 @@ static int
 __live_restore_populate_queue(WT_SESSION_IMPL *session, uint64_t *work_count)
 {
     WT_DECL_RET;
-    WT_LIVE_RESTORE_SERVER *server = S2C(session)->live_restore_server;
+    WT_CONNECTION_IMPL *conn = S2C(session);
+    WT_LIVE_RESTORE_SERVER *server = conn->live_restore_server;
 
+    /* Initialize the work queue. */
     TAILQ_INIT(&server->work_queue);
-    /*
-     * Open a metadata cursor to gather the list of objects. The metadata file is built from the
-     * WiredTiger.backup file, during __wt_turtle_init. Thus this function must be run after that
-     * function.
-     */
+    __wt_verbose_debug1(
+      session, WT_VERB_FILEOPS, "%s", "Live restore server: Initializing the work queue");
 
     /*
-     * FIXME-WT-13888: Add logic to queue log files first, then the history store. This will use a
-     * directory list call.
+     * The work queuing algorithm is as follows, note all items are queued at the head:
+     *     - First we walk the metadata file to gather a list of file: objects. This will include
+     *     the history store file. We don't define any special ordering for it.
+     *     - If logging is enabled all the log files are found and queued.
+     *     - If we are not restoring from a backup, which is possible with live restore we queue
+     *     the metadata file.
      */
+
     WT_CURSOR *cursor;
     WT_RET(__wt_metadata_cursor(session, &cursor));
-    WT_LIVE_RESTORE_WORK_ITEM *work_item = NULL;
-    __wt_verbose_debug1(
-      session, WT_VERB_FILEOPS, "%s", "Live restore server: Initializing the work queue");
-
+    /* Early declaration because of WT_ERR usage. */
```

Comment:
Given we're queuing using the metadata file is there a chance that log replay needs to happen before this? i.e. does the metadata file always contain the full list of objects without needing to replay logs? Which is simply asking do we log metadata file operations?